generators/Exercises/Diamond.cs

```diff
@@ -0,0 +1,6 @@
+﻿namespace Generators.Exercises
+{
+    public class Diamond : CustomExercise
+    {
```

Comment:
It's probably worth adding a comment here on why we decided not to follow the canonical data for this exercise (here's it's because we wanted to do property based testing). That way if something changes in the future and we want to revisit that decision we have the info here without digging through the issues.

Reply:
Good point. I've updated the PR.